Sources/NIOCertificateHotReloading/NIOCertificateHotReloading.swift

```
@@ -0,0 +1,319 @@
+    /// private key pair.
+    /// - Important: You *must* call this method to get certificate and key updates.
+    public func run() async throws {
+        while !Task.isCancelled {
+            try await Task.sleep(nanoseconds: UInt64(self.refreshInterval.nanoseconds))
+            self.reloadPair()
+        }
+    }
+
+    private func reloadPair() {
+        if let certificateBytes = self.loadCertificate(),
+            let keyBytes = self.loadPrivateKey(),
+            let certificate = self.parseCertificate(from: certificateBytes),
+            let key = self.parsePrivateKey(from: keyBytes),
+            key.publicKey.isValidSignature(certificate.signature, for: certificate)
+        {
+            self.attemptToUpdatePair(certificate: certificate, key: key)
+        }
+    }
+
+    private func loadCertificate() -> [UInt8]? {
+        let certificateBytes: [UInt8]?
+        switch self.certificateDescription.location._backing {
+        case .file(let path):
+            let bytes = FileManager.default.contents(atPath: path)
+            certificateBytes = bytes.map { Array($0) }
+
+        case .memory(let bytesProvider):
+            certificateBytes = bytesProvider()
+        }
+        return certificateBytes
+    }
+
+    private func loadPrivateKey() -> [UInt8]? {
+        let keyBytes: [UInt8]?
+        switch self.privateKeyDescription.location._backing {
+        case .file(let path):
+            let bytes = FileManager.default.contents(atPath: path)
+            keyBytes = bytes.map { Array($0) }
+
+        case .memory(let bytesProvider):
+            keyBytes = bytesProvider()
+        }
+        return keyBytes
+    }
+
+    private func parseCertificate(from certificateBytes: [UInt8]) -> Certificate? {
+        let certificate: Certificate?
+        switch self.certificateDescription.format._backing {
+        case .der:
+            certificate = try? Certificate(derEncoded: certificateBytes)
```

We should let the error be thrown out and at least logged (here and all the `try?` below as well)

---

I am not sure about throwing - I don't think we want the reloader to stop running when some potentially transient error is encountered. 

We could have a max number of failed attempts though, and throw after that number has been reached.

W.r.t logging, we don't currently depend on `swift-log` in this package, so I'm not sure if that's something we want. I guess we could also just `print` (which I know is not ideal but it's what we already do for the [debug events handler](https://github.com/apple/swift-nio-extras/blob/f1f6f772198bee35d99dd145f1513d8581a54f2c/Sources/NIOExtras/DebugInboundEventsHandler.swift#L155))

---

Okay so given the other discussion about having throwing inits, I _have_ made these methods throw, but I am just ignoring the errors from the main reload loop.

Still open questions about having a max number of failed attempts and a logger.